modules/extensions-spring-support/src/main/java/org/mule/module/extension/internal/capability/xml/schema/SchemaBuilder.java

```diff
@@ -0,0 +1,812 @@
+    {
+        Import xmlImport = new Import();
+        xmlImport.setNamespace(SchemaConstants.XML_NAMESPACE);
+        schema.getIncludeOrImportOrRedefine().add(xmlImport);
+        return this;
+    }
+
+    private SchemaBuilder importSpringFrameworkNamespace()
+    {
+        Import springFrameworkImport = new Import();
+        springFrameworkImport.setNamespace(SchemaConstants.SPRING_FRAMEWORK_NAMESPACE);
+        springFrameworkImport.setSchemaLocation(SchemaConstants.SPRING_FRAMEWORK_SCHEMA_LOCATION);
+        schema.getIncludeOrImportOrRedefine().add(springFrameworkImport);
+        return this;
+    }
+
+    private SchemaBuilder importMuleNamespace()
+    {
+        Import muleSchemaImport = new Import();
+        muleSchemaImport.setNamespace(SchemaConstants.MULE_NAMESPACE);
+        muleSchemaImport.setSchemaLocation(SchemaConstants.MULE_SCHEMA_LOCATION);
+        schema.getIncludeOrImportOrRedefine().add(muleSchemaImport);
+        return this;
+    }
+
+    public SchemaBuilder registerConfigElement(final Configuration configuration)
+    {
+        Map<QName, String> otherAttributes = new HashMap<>();
+        final ExtensionType config = registerExtension(configuration.getName(), otherAttributes);
+        config.getAttributeOrAttributeGroup().add(createNameAttribute());
+
+        final ExplicitGroup choice = new ExplicitGroup();
+        choice.setMinOccurs(new BigInteger("0"));
+        choice.setMaxOccurs("unbounded");
+        config.setChoice(choice);
+
+
+        for (final Parameter parameter : configuration.getParameters())
+        {
+            parameter.getType().getQualifier().accept(new BaseDataQualifierVisitor()
+            {
+
+                private boolean forceOptional = false;
+
+                @Override
+                public void onList()
+                {
+                    forceOptional = true;
+                    defaultOperation();
+                    generateCollectionElement(choice, parameter, true);
+                }
+
+                @Override
+                public void onPojo()
+                {
+                    forceOptional = false;
+                    defaultOperation();
+                    registerComplexTypeChildElement(choice,
+                                                    parameter.getName(),
+                                                    parameter.getDescription(),
+                                                    parameter.getType(),
+                                                    isRequired(parameter, forceOptional));
+                }
+
+                @Override
+                protected void defaultOperation()
+                {
+                    config.getAttributeOrAttributeGroup().add(createAttribute(parameter, isRequired(parameter, forceOptional)));
+                }
+            });
+        }
+
+        config.setAnnotation(createDocAnnotation(configuration.getDescription()));
+
+        if (choice.getParticle().size() == 0)
```

Comment:
the purpose of this is to remove the choice element if the config requires no nested elements. I realise this is not obvious so I changed it to only add the choice only when needed instead of adding it and removing it